Validation/SiTrackerPhase2V/plugins/Phase2ITValidateRecHit.cc

```diff
@@ -132,16 +132,19 @@ void Phase2ITValidateRecHit::fillITHistos(const edm::Event& iEvent,
       //GetSimHits
       const std::vector<SimHitIdpr>& matchedId = associateRecHit.associateHitId(rechit);
       const PSimHit* simhitClosest = nullptr;
-      float minx = 10000;
+      float mind = 1e4;
       LocalPoint lp = rechit.localPosition();
       for (const auto& simHitCol : simHits) {
         for (const auto& simhitIt : *simHitCol) {
           if (detId.rawId() != simhitIt.detUnitId())
             continue;
           for (const auto& mId : matchedId) {
             if (simhitIt.trackId() == mId.first) {
-              if (!simhitClosest || abs(simhitIt.localPosition().x() - lp.x()) < minx) {
-                minx = abs(simhitIt.localPosition().x() - lp.x());
+              float dx = simhitIt.localPosition().x() - lp.x();
```

Comment:
I think this whole block of finding the closest simhit is repeated in several classes. Perhaps can be moved to the base or even higher up.